lib/wpc_proto/internal_modules/proto_otap.c

```diff
@@ -0,0 +1,115 @@
+/* Wirepas Oy licensed under Apache License, Version 2.0
+ *
+ * See file LICENSE for full license details.
+ *
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "wpc.h"
+#include "proto_otap.h"
+#include "proto_config.h"
+#include <pb_encode.h>
+#include <pb_decode.h>
+#include "platform.h"
+#include "common.h"
+
+#define LOG_MODULE_NAME "otap_proto"
+#define MAX_LOG_LEVEL INFO_LOG_LEVEL
+#include "logger.h"
+
+static app_scratchpad_status_t otap_status = {0};
+
+static bool initialize_otap_variables()
+{
+    if (WPC_get_local_scratchpad_status(&otap_status) != APP_RES_OK)
+    {
+        LOGE("Cannot get local scratchpad status\n");
+        return false;
+    }
+    return true;
+}
+
+bool Proto_otap_init(void)
+{
+    /* Read initial config from sink */
+    if (!initialize_otap_variables())
```

Comment:
What is the point to read it initially if at the end it is read for every request?

Reply:
As in sink service. And as it's in the config, it's not updated each time we need it. The question is also : do we need to update it each time, or can we rely on the cache config ?